logstash-core/src/main/java/org/logstash/ext/JrubyWrappedSynchronousQueueExt.java

```diff
@@ -0,0 +1,47 @@
+package org.logstash.ext;
+
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyNumeric;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+@JRubyClass(name = "WrappedSynchronousQueue")
+public class JrubyWrappedSynchronousQueueExt extends RubyObject {
+
+    private BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue;
+
+    public JrubyWrappedSynchronousQueueExt(final Ruby runtime, final RubyClass metaClass) {
+        super(runtime, metaClass);
+    }
+
+    @JRubyMethod(name = "initialize")
+    @SuppressWarnings("unchecked")
+    public void rubyInitialize(final ThreadContext context, IRubyObject size) {
+        int typedSize = ((RubyNumeric)size).getIntValue();
+        this.queue = new ArrayBlockingQueue<>(typedSize);
+    }
+
+    @JRubyMethod(name = "write_client")
+    public IRubyObject getWriteClient(final ThreadContext context) {
+        return JrubyMemoryWriteClientExt.create(queue);
+    }
+
+    @JRubyMethod(name = "read_client")
+    public IRubyObject getReadClient(final ThreadContext context) {
+        return JrubyMemoryReadClientExt.create(queue, 125, 50);
```

Comment:
These numbers come directly from the class you're replacing, but it would be helpful to add a comment indicating why they are what they are (even if doing so requires a little archaeology).

Reply:
I found #8707 in which those values were introduced and added a comment to that effect.